src/test/rpc/AccountCurrencies_test.cpp

```diff
@@ -0,0 +1,175 @@
+        std::generate (gwCurrencies.begin(), gwCurrencies.end(),
+            [&]()
+            {
+                auto gwc = gw[std::string("US") + currencySuffix++];
+                env (trust (alice, gwc (100)));
+                return gwc;
+            });
+        env.close ();
+
+        Json::Value params;
+        params[jss::account] = alice.human();
+        auto result = env.rpc ("json", "account_currencies",
+            boost::lexical_cast<std::string>(params)) [jss::result];
+
+        auto arrayCheck =
+            [&result] (
+                Json::StaticString const& fld,
+                std::vector<boost::optional<IOU>> const& expected) -> bool
+            {
+                bool stat =
+                    result.isMember (fld) &&
+                    result[fld].isArray() &&
+                    result[fld].size() == expected.size();
+                for (size_t i = 0; stat && i < expected.size(); ++i)
+                {
+                    Currency foo;
+                    stat &= (
+                        to_string(expected[i].value().currency) ==
+                        result[fld][i].asString()
+                    );
+                }
+                return stat;
+            };
+
+        BEAST_EXPECT (arrayCheck (jss::receive_currencies, gwCurrencies));
+        BEAST_EXPECT (arrayCheck (jss::send_currencies, {}));
+
+        // now form a payment for each currency
+        for (auto const& c : gwCurrencies)
+            env (pay (gw, alice, c.value()(50)));
+
+        result = env.rpc ("json", "account_currencies",
+            boost::lexical_cast<std::string>(params)) [jss::result];
+
+        // send_currencies should be populated now
+        BEAST_EXPECT (arrayCheck (jss::receive_currencies, gwCurrencies));
+        BEAST_EXPECT (arrayCheck (jss::send_currencies, gwCurrencies));
+
+        // make a payment that exhausts the trustline from alice to gw for USA
+        env (pay (gw, alice, gw["USA"](50)));
+        result = env.rpc ("json", "account_currencies",
+            boost::lexical_cast<std::string>(params)) [jss::result];
+
+        // USA should now be missing from receive_currencies
+        decltype(gwCurrencies) gwCurrenciesNoUSA (gwCurrencies.begin() + 1,
+            gwCurrencies.end());
+        BEAST_EXPECT (arrayCheck (jss::receive_currencies, gwCurrenciesNoUSA));
+        BEAST_EXPECT (arrayCheck (jss::send_currencies, gwCurrencies));
+
+        // add trust from gw to alice and then exhaust that trust line
+        // so that send_currencies for alice will now omit USA
+        env (trust (gw, alice["USA"] (100)));
+        env (pay (alice, gw, alice["USA"](200)));
+        result = env.rpc ("json", "account_currencies",
+            boost::lexical_cast<std::string>(params)) [jss::result];
+        BEAST_EXPECT (arrayCheck (jss::receive_currencies, gwCurrencies));
+        BEAST_EXPECT (arrayCheck (jss::send_currencies, gwCurrenciesNoUSA));
+    }
```

Comment:
https://ripple.com/build/rippled-apis/#account-currencies
>This method doesn't check whether the trust line is frozen or authorized.
What do you think about testing that ^?